src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -188,624 +253,706 @@ private static boolean isFieldInitializedInTheConstructor(HaxeFieldModel field)
-    final String actualPackage2 = HaxeResolveUtil.getPackageName(element.getContainingFile());
-    // @TODO: Should use HaxeResolveUtil
+  @Override
+  public void visitAssignExpression(@NotNull HaxeAssignExpression element) {
+    final PsiElement left = element.getFirstChild();
+    final PsiElement right = element.getLastChild();
+
+    if (left != null && right != null) {
+      final ResultHolder leftResult = HaxeTypeResolver.getPsiElementType(left);
+      final ResultHolder rightResult = HaxeTypeResolver.getPsiElementType(right);
 
-    for (String s : StringUtils.split(packageName, '.')) {
-      if (!s.substring(0, 1).toLowerCase().equals(s.substring(0, 1))) {
-        //HaxeSemanticError.addError(element, new HaxeSemanticError("Package name '" + s + "' must start with a lower case character"));
-        // @TODO: Move to bundle
-        holder.createErrorAnnotation(element, "Package name '" + s + "' must start with a lower case character");
+      if (leftResult.isUnknown()) {
+        leftResult.setType(rightResult.getType());
       }
-    }
+      leftResult.removeConstant();
 
-    if (!packageName.equals(actualPackage)) {
-      holder.createErrorAnnotation(
-        element,
-        "Invalid package name! '" + packageName + "' should be '" + actualPackage + "'").registerFix(
-        new HaxeFixer("Fix package") {
-          @Override
-          public void run() {
-            Document document =
-              PsiDocumentManager.getInstance(element.getProject()).getDocument(element.getContainingFile());
+      if (!leftResult.canAssign(rightResult)) {
+        annotateNotCompatibleTypes(right, leftResult, rightResult, true);
+      }
+    }
+  }
 
-            if (expression != null) {
-              TextRange range = expression.getTextRange();
-              document.replaceString(range.getStartOffset(), range.getEndOffset(), actualPackage);
-            } else {
-              int offset =
-                element.getNode().findChildByType(HaxeTokenTypes.OSEMI).getTextRange().getStartOffset();
-              document.replaceString(offset, offset, actualPackage);
-            }
-          }
-        }
-      );
+  @Override
+  public void visitStringLiteralExpression(@NotNull HaxeStringLiteralExpression o) {
+    if (isSingleQuotesRequired(o)) {
+      holder.createWarningAnnotation(o, HaxeBundle.message("haxe.semantic.string.interpolation.quotes"));
     }
   }
-}
 
-class MethodBodyChecker {
-  public static void check(HaxeMethod psi, AnnotationHolder holder) {
-    final HaxeMethodModel method = psi.getModel();
-    HaxeTypeResolver.getPsiElementType(method.getBodyPsi(), holder);
+  private static boolean isSingleQuotesRequired(HaxeStringLiteralExpression o) {
+    if (o.getParent() instanceof HaxeObjectLiteralElement) return false;
+    return (o.getLongTemplateEntryList().size() > 0 || o.getShortTemplateEntryList().size() > 0) &&
+           o.getFirstChild().textContains('"');
```

Comment:
Use `o.getFirstChild().getText().startsWith('"');`. Otherwise, escaped quotes will be detected.